turbinia/workers/volatility.py

```diff
@@ -59,23 +60,40 @@ def run(self, evidence, result):
             evidence.local_path, evidence.profile, self.module,
             output_file_path).split()
 
-    result.log('Running vol as [{0:s}]'.format(' '.join(cmd)))
-    res = self.execute(
-        cmd, result, log_files=[debug_log], new_evidence=[output_evidence],
-        close=True)
+    result.log('Running volatility as [{0:s}]'.format(' '.join(cmd)))
+    res = self.execute(cmd, result, new_evidence=[output_evidence], close=True)
 
     if res == 0:
-      # Write the report to the output file.
-      with open(output_file_path, 'w') as fh:
-        fh.write(output_evidence.text_data.encode('utf-8'))
-
-      # Add the resulting evidence to the result object.
-      result.add_evidence(output_evidence, evidence.config)
-      result.report_data = output_evidence
-      summary = 'Volatility module {0} successfully ran.'.format(self.module)
+      # Get report data from the output file.
+      try:
+        file_size = os.stat(output_file_path).st_size
+      except OSError as exception:
+        msg = 'Cannot read output file {0:s}: {1!s}'.format(
+            output_file_path, exception)
+        summary = 'Volatility ran successfully, but no output file was created'
+        result.log(msg)
+        result.close(self, success=False, status=summary)
+        return result
+
+      if file_size > MAX_REPORT_SIZE:
+        result.log(
+            'Volatility report output size ({0:d}) is greater than max report '
+            'size ({1:d}). Truncating report to max size'.format(
+                file_size, MAX_REPORT_SIZE))
+        summary = (
+            'Volatility module {0:s} successfuly ran (report truncated)'.format(
+                self.module))
+      else:
+        summary = 'Volatility module {0:s} successfully ran'.format(self.module)
+
+      with open(output_file_path, 'rb') as fh:
+        output_evidence.text_data = codecs.decode(
```

Comment:
I opt to now do decode and read on the same line this makes debugging harder
```
report_data = fh.read(MAX_REPORT_SIZE)
output_evidence.text_data = report_data.decode('utf-8')
```
also there is no need to use codecs and make sure to catch UnicodeDecodeErrors

Reply:
or alternatively open with encoding (python 3 only) of io.open with encoding (python 2 and 3 compatible)

```
with io.open(output_file_path, 'r', encoding='utf-8') as fh:
  output_evidence.text_data = fh.read(MAX_REPORT_SIZE)
```

Reply:
Ok -- I split this into two lines, but I didn't use io.open() for now.  Once python2 is completely removed (planned for the release after next), I'll take a pass and upgrade some things like this.